src/chainlit/playground/providers/langchain.py

```diff
@@ -0,0 +1,79 @@
+from typing import List, Union
+
+from fastapi.responses import PlainTextResponse
+from langchain.schema.messages import AIMessage, HumanMessage, SystemMessage
+
+from chainlit import input_widget
+from chainlit.playground.provider import BaseProvider
+from chainlit.prompt import PromptMessage
+
+
+class LangchainGenericProvider(BaseProvider):
+    from langchain.chat_models.base import BaseChatModel
+    from langchain.llms.base import LLM
+
+    llm: Union[LLM, BaseChatModel]
+
+    def __init__(
+        self,
+        id: str,
+        name: str,
+        llm: Union[LLM, BaseChatModel],
+        inputs: List[input_widget.InputWidget],
+        is_chat: bool = False,
+    ):
+        super().__init__(
+            id=id,
+            name=name,
+            env_vars={},
+            inputs=inputs,
+            is_chat=is_chat,
+        )
+        self.llm = llm
+
+    def prompt_message_to_langchain_message(self, message: PromptMessage):
+        content = "" if message.formatted is None else message.formatted
+        if message.role == "user":
+            return HumanMessage(content=content)
+        elif message.role == "assistant":
+            return AIMessage(content=content)
+        elif message.role == "system":
+            return SystemMessage(content=content)
+        else:
+            raise ValueError(f"Got unknown type {message}")
+
+    def format_message(self, message, prompt):
+        message = super().format_message(message, prompt)
+        return message
+
+    def message_to_string(self, message: PromptMessage) -> str:
+        return message.to_string()
+
+    async def create_completion(self, request):
+        await super().create_completion(request)
+
+        # Update the request.prompt.messages with the formatted messages
+        self.create_prompt(request)
+
+        if self.is_chat:
+            result = await self.llm._agenerate(
+                messages=[
+                    self.prompt_message_to_langchain_message(i)
+                    for i in request.prompt.messages
+                ],
+                **request.prompt.settings,
+            )
+
+            return PlainTextResponse(
+                result.generations[0].text, media_type="text/plain"
```

Comment:
we should support streaming if the llm supports streaming. Also is it working to send a PlainTextResponse if the frontend expects an event stream?

Reply:
we do support streaming now 👌